lib/javascript/JavascriptModulesPlugin.js

```diff
@@ -1420,25 +1432,37 @@ class JavascriptModulesPlugin {
 	 * @param {Set<Module>} inlinedModules inlinedModules
 	 * @param {ChunkRenderContext} chunkRenderContext chunkRenderContext
 	 * @param {CompilationHooks} hooks hooks
-	 * @returns {Map<Module, Source>} renamed inlined modules
+	 * @param {boolean} allStrict allStrict
+	 * @param {boolean} hasChunkModules hasChunkModules
+	 * @returns {Map<Module, Source> | false} renamed inlined modules
 	 */
-	renameInlineModule(
+	getRenamedInlineModule(
 		allModules,
 		renderContext,
 		inlinedModules,
 		chunkRenderContext,
-		hooks
+		hooks,
+		allStrict,
+		hasChunkModules
 	) {
+		const innerStrict = !allStrict && allModules.every(m => m.buildInfo.strict); // only the single entry for now.
+		const singleEntryWithModules = inlinedModules.size === 1 && hasChunkModules;
```

Comment:
Can you describe it? Why only the single entry for now?

Also let's add a test case for it

Reply:
Now the only single entry works, see https://github.com/webpack/webpack/blob/981943db89223ab1bcb609e539599430c120c894/lib/javascript/JavascriptModulesPlugin.js#L867-L873, because only optimization for single entry is currently done. This code added here is to ensures the optimization is only done when there is a single entry reason and avoids optimization being wrapped in IIFE for other reasons.

Test case has been added in https://github.com/webpack/webpack/blob/main/test/cases/entry-inline/no-var-leak/index.js#L3 and https://github.com/webpack/webpack/blob/main/test/configCases/output-module/inlined-module/index.js, what specific test case we can add further? 👀